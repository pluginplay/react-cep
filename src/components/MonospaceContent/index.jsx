import React from 'react'
import { FlexItem } from 'styled-flex-component'
import styled from 'styled-components'

class MonospaceContent extends React.Component {

  onClick (e) {
    e.preventDefault()
    const range = document.createRange()
    const sel = window.getSelection()
    range.selectNodeContents(e.target)
    sel.removeAllRanges()
    sel.addRange(range)
  }

  render () {
    return (
      <FlexItem {...this.props} onClick={this.onClick}>
        {this.props.children}
      </FlexItem>
    )
  }

}

export default styled(MonospaceContent)`
  background: ${props => props.theme.background.darker};
  border-radius: 2px;
  padding: 5px;
  margin: 0;
  width: 100%;
  box-sizing: border-box;
  font-family: monospace;
  overflow: auto;
  text-align: left;
`
