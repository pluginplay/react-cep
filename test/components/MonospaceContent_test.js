import { expect, withTheme } from '../setup'
import renderer from 'react-test-renderer'
import MonospaceContent from '../../src/components/MonospaceContent'
import { FlexItem } from 'styled-flex-component'
import { mount } from 'enzyme'

it('renders correctly', () => {
  const tree = renderer
    .create(withTheme(<MonospaceContent>test text</MonospaceContent>))
    .toJSON()
  jestExpect(tree).toMatchSnapshot()
})

it('forwards properties to FlexItem', () => {
  const wrapper = mount(withTheme(<MonospaceContent grow={1}>test text</MonospaceContent>))
  const flexItem = wrapper.find(FlexItem).first()
  expect(flexItem).to.have.prop('grow').equal(1)
  expect(flexItem).to.have.text('test text')
  expect(flexItem).to.have.prop('onClick')
})
