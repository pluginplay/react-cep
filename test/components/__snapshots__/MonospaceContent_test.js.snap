// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
.c1 {
  -webkit-order: 0;
  -ms-flex-order: 0;
  order: 0;
  -webkit-flex-basis: auto;
  -ms-flex-preferred-size: auto;
  flex-basis: auto;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  display: block;
}

.c0 {
  background: #0c0c0c;
  border-radius: 2px;
  padding: 5px;
  margin: 0;
  width: 100%;
  box-sizing: border-box;
  font-family: monospace;
  overflow: auto;
  text-align: left;
}

<div
  className="c0 c1"
  onClick={[Function]}
>
  test text
</div>
`;
